roles/openshift_control_plane/tasks/ensure_nodes_matching_selector.yml

```diff
@@ -1,4 +1,17 @@
 ---
+- name: Check that the sync daemonset is ready so labels will be up to date
+  oc_obj:
+    state: list
+    kind: daemonset
+    name: sync
+    namespace: openshift-node
+  register: __status_of_sync_ds
```

Comment:
It should keep fetching sync status until DS is ready at least:

```
until:
  - __status_of_sync_ds.results is defined
  - __status_of_sync_ds.results.results is defined
  - __status_of_sync_ds.results.results | length > 0
  - __status_of_sync_ds.results.results[0].status is defined
  - __status_of_sync_ds.results.results[0].status.numberAvailable is defined
  - __status_of_sync_ds.results.results[0].status.desiredNumberScheduled is defined
  - __status_of_sync_ds.results.results[0].status.numberAvailable > __status_of_sync_ds.results.results[0].status.desiredNumberScheduled
```

No need to set additional facts and ignore errors - sync must be run eventually

Reply:
Thanks for the feedback, I'll update the PR.  I had considered doing something like this, but was concerned with slowing down the install process since the bz said it took about 11 min for the nodes to be labelled in the test environment.

Reply:
Should the last comparison check should be "numberAvailable == desiredNumberScheduled"?  I didn't think there were any cases there the numberAvailable would be greater than the number scheduled.

Reply:
`==` is fine, can't imagine a situation where numberAvailable is larger